### DISCUSSION TITLE
Mesh picking returns polydata and not actor

### DISCUSSION THREAD

**Discussion by HametAl** [pyvista/pyvista#4066] [Q&A] [answered]
I am trying to select an actor and then delete it, but when I use enable_mesh_picking, it returns a polyData instead of an actor:

......
......
........
#here I create the actor 
window.surf=pv.PolyData()
        window.surf.points=window.points
        if window.surf.n_points>2:
            window.surf.faces=np.hstack([np.array([window.surf.n_points,0]),np.arange(1,window.surf.n_points,1)])
            window.plotter.add_mesh(window.surf,name=f'_surf{self.counter}',show_edges=True,point_size=10)
........
........
......
#here I select it

def selected(actor):
        window.plotter.remove_actor(actor)
window.plotter.enable_mesh_picking(selected,use_actor=True,show=False,left_clicking=True)

But I get this error:
TypeError: RemoveActor argument 1: method requires a vtkProp, a vtkPolyData was provided.

**Comment by banesullivan** [comment]
Please share a report:

```py
import pyvista
print(pyvista.Report())
```

**Reply by HametAl** [reply, marked as answer]
My pyvista version was 37. I updated 38.3 and the problem is solved now. Thank you!
